extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -24,8 +24,8 @@ export interface ILocalSettingsJson {
 }
 
 export interface IFileFunctionObject {
-	filePromise: Promise<string>;
-	watcherDisposable: vscode.Disposable;
+	filePromise: Promise<string> | undefined;
```

Comment:
Why would these ever need to be undefined?

Reply:
Need to have this undefined since in the `finally block` we dispose of it and that would happen if the user exits out of selecting a folder. This would make the `newFunctionFileObject` undefined.

Reply:
Then just make the whole object undefined - not the specific properties within it. I don't see any reason why either of these would ever be undefined - they're created at the same time and once they're done being used the entire object can just be deleted.